### ISSUE TITLE
Async Encoding

### ISSUE THREAD

**Issue by DjDeveloperr** [DjDeveloperr/skia_canvas#28]
Just need to offload encoding to a separate thread and call with non blocking: true. It'll be useful for applications which handle concurrent requests and render images for them.

Also see: #27

**Comment by lino-levan** [comment]
Implementation-wise, this should be done with web workers right?

**Comment by DjDeveloperr** [comment]
Nope, workers aren't needed. We just need to add another FFI symbol definition that refers to same encoding symbol but with nonblocking: true.
